opensrp-opd/src/main/java/org/smartregister/opd/adapter/ProfileOverviewAdapter.java

```diff
@@ -0,0 +1,259 @@
+package org.smartregister.opd.adapter;
+
+import android.app.Activity;
+import android.content.Context;
+import android.support.annotation.NonNull;
+import android.support.v7.app.AlertDialog;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.text.TextUtils;
+import android.view.Gravity;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import org.jeasy.rules.api.Facts;
+import org.smartregister.opd.OpdLibrary;
+import org.smartregister.opd.R;
+import org.smartregister.opd.contract.PreviousContactsTests;
+import org.smartregister.opd.domain.YamlConfigItem;
+import org.smartregister.opd.domain.YamlConfigWrapper;
+import org.smartregister.opd.utils.OpdUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by Ephraim Kigamba - ekigamba@ona.io on 2019-09-30
+ */
+public class ProfileOverviewAdapter extends RecyclerView.Adapter<ProfileOverviewAdapter.ViewHolder> {
+
+    private List<YamlConfigWrapper> mData;
+    private LayoutInflater mInflater;
+    private Facts facts;
+    private Context context;
+    //private AllTestClickListener allTestClickListener = new AllTestClickListener();
+    private PreviousContactsTests.Presenter presenter;
+    private String baseEntityId;
+    private String contactNo;
+
+    // data is passed into the constructor
+    public ProfileOverviewAdapter(Context context, List<YamlConfigWrapper> data, Facts facts) {
```

Comment:
Check NonNull or/and Nullable annnotations, there is alot of commented out code and some dead code